### PR TITLE
fix(testing): Add sorting for testing dest migrations

### PR DESCRIPTION
**Pull request by yevgenypats**
This should fix some flakey tests for destination and also fix germlin - https://github.com/cloudquery/cloudquery/pull/10188

### CODE REVIEW

**Review by hermanschaaf — APPROVED** [review]

File: testdata/testdata.go [hunk 1]
```diff
@@ -243,6 +251,14 @@ func GenTestData(sc *arrow.Schema, opts GenTestDataOptions) []arrow.Record {
 		records = append(records, bldr.NewRecord())
 		bldr.Release()
 	}
+	if indices := sc.FieldIndices(schema.CqIDColumn.Name); len(indices) > 0 {
```
Comment (disq) [inline testdata/testdata.go:254]: Why is this needed?

**Review by disq — APPROVED** [review]
lgtm otherwise
